lib/active_model/serializer/array_serializer.rb

```diff
@@ -21,11 +22,11 @@ def initialize(objects, options = {})
       end
 
       def json_key
-        @objects.first.json_key if @objects.first
-      end
-
-      def root=(root)
-        @objects.first.root = root if @objects.first
+        if root == true && @objects.first
+          @objects.first.class.root_name
```

Comment:
I'm not sure about using the ```root_name``` here. ArraySerializer is pretty abstract, take a lott at this piece of code for example:

```ruby
@objects = objects.map do |object|
  serializer_class = options.fetch(
    :serializer,
    ActiveModel::Serializer.serializer_for(object)
  )
  serializer_class.new(object, options.except(:serializer))
end
```

By checking this code we can presume that the objects might not be instances from the same class. Wrap them by the class name from its first object doesn't would be right.